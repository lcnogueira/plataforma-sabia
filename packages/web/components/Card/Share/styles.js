import styled, { css, keyframes } from 'styled-components';
import { AiOutlineShareAlt } from 'react-icons/ai';

const zoom = keyframes`
	0%, 100% {
		transform: scale(1);
	}
	50% {
		transform: scale(1.1);
	}
`;

export const Container = styled.button`
	${({ theme: { sizes } }) => css`
		display: flex;
		align-items: center;
		justify-content: space-between;
		padding: 0.5rem;
		gap: 0.6rem;
		cursor: pointer;
		background: transparent;
		border: none;

		svg {
			width: ${sizes.defaultIcon}rem;
			height: ${sizes.defaultIcon}rem;

			&:hover {
				animation: ${zoom} 1.5s;
			}
		}
	`}
`;

export const Icon = styled(AiOutlineShareAlt).attrs(({ theme: { colors } }) => ({
	color: colors.lightGray2,
}))``;
