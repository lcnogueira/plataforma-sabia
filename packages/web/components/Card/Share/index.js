import React from 'react';
import PropTypes from 'prop-types';
import { useModal } from '../../../hooks';
import { Container, Icon } from './styles';

const Share = ({ id }) => {
	const { openModal } = useModal();

	return (
		<Container onClick={() => openModal('share', { id })}>
			<Icon />
		</Container>
	);
};

Share.propTypes = {
	id: PropTypes.number.isRequired,
};

export default Share;
