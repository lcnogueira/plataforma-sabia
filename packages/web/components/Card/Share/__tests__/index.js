import React from 'react';
import { render, fireEvent } from 'test-utils';
import Share from '..';

describe('ShareModal component', () => {
	it('should open the sharing modal when clicking', () => {
		window.open = jest.fn();
		document.execCommand = jest.fn();

		const { container, getByRole, getByLabelText } = render(<Share />);

		const shareButton = getByRole('button', { name: 'share' });
		fireEvent.click(shareButton);

		const iconButtons = ['facebook', 'clipboard'];
		iconButtons.forEach((button) => {
			const clipboardButton = getByLabelText(button);
			fireEvent.click(clipboardButton);
		});

		expect(window.open).toHaveBeenCalledTimes(1);
		expect(window.open).toHaveBeenCalledWith(
			expect.stringMatching(/^[https://facebook.com/sharer/sharer.php?u=(.*)]/),
			'_blank',
			'noopener',
		);
		expect(document.execCommand).toHaveBeenCalledTimes(1);
		expect(container).toMatchSnapshot();
	});
});
