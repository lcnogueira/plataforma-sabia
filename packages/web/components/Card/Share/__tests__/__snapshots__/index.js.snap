// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareModal component should open the sharing modal when clicking 1`] = `
.c0 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 1500;
  background: rgba(1,1,1,0.5);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: auto;
}

.c1 {
  position: fixed;
  min-width: 40rem;
  max-height: 100vh;
  z-index: 2000;
  padding: 4rem;
  background: hsla(0,0%,100%);
  border-radius: 0.5rem;
  box-shadow: 0px 0px 20px -5px rgba(38,38,38,1);
}

.c2 {
  background: 0;
  border: 0;
  position: absolute;
  top: 1rem;
  right: 1rem;
}

.c2 svg {
  height: 3rem;
  width: 3rem;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
}

.c2 svg:hover {
  color: hsla(29,94%,62%);
}

.c5 {
  background-color: hsla(29,94%,62%);
  color: hsla(0,0%,100%);
  border-radius: 0.5rem;
  border: none;
  font-size: 2.2rem;
  text-transform: none;
  padding: 1.8rem 6rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
}

.c5:hover {
  opacity: 0.8;
}

.c6 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  opacity: 1;
}

.c6::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c6::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c6:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c6::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c3 {
  width: 50rem;
  padding: 2rem;
}

.c3 h3 {
  font-size: 1.8rem;
  text-align: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 2rem 0;
}

.c4 button {
  margin: 1rem;
  padding: 1rem;
}

.c7 {
  margin: 0;
  text-align: center;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.5rem;
  gap: 0.6rem;
  cursor: pointer;
  background: transparent;
  border: none;
}

.c8 svg {
  width: 3rem;
  height: 3rem;
}

.c8 svg:hover {
  -webkit-animation: jBAKzd 1.5s;
  animation: jBAKzd 1.5s;
}

@media (max-width:768px) {
  .c1 {
    min-width: 90%;
    padding: 2rem;
    margin: 1rem;
    overflow-x: hidden;
    overflow-y: scroll;
  }
}

@media (max-width:768px) {
  .c5 {
    font-size: 1.6rem;
    padding: 1.4rem 6rem;
  }
}

@media (max-width:768px) {
  .c3 {
    width: 100%;
  }
}

@media (max-width:450px) {
  .c4 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:450px) {
  .c4 button {
    width: 100%;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      data-testid="modal"
    >
      <button
        aria-label="Close modal"
        class="c2"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 0 1-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
          />
        </svg>
      </button>
      <div
        class="c3"
      >
        <h3>
          common:shareTechnology
        </h3>
        <div
          class="c4"
        >
          <button
            aria-label="facebook"
            class="styles__IconButton-sc-12qr9fv-3 c5"
            color="hsla(0, 0%, 100%)"
          >
            <svg
              class="styles__FacebookIcon-sc-12qr9fv-4"
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-92.4 233.5h-63.9c-50.1 0-59.8 23.8-59.8 58.8v77.1h119.6l-15.6 120.7h-104V912H539.2V602.2H434.9V481.4h104.3v-89c0-103.3 63.1-159.6 155.3-159.6 44.2 0 82.1 3.3 93.2 4.8v107.9z"
              />
            </svg>
          </button>
          <button
            aria-label="twitter"
            class="styles__IconButton-sc-12qr9fv-3 c5"
            color="hsla(0, 0%, 100%)"
          >
            <svg
              class="styles__TwitterIcon-sc-12qr9fv-5"
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M928 254.3c-30.6 13.2-63.9 22.7-98.2 26.4a170.1 170.1 0 0 0 75-94 336.64 336.64 0 0 1-108.2 41.2A170.1 170.1 0 0 0 672 174c-94.5 0-170.5 76.6-170.5 170.6 0 13.2 1.6 26.4 4.2 39.1-141.5-7.4-267.7-75-351.6-178.5a169.32 169.32 0 0 0-23.2 86.1c0 59.2 30.1 111.4 76 142.1a172 172 0 0 1-77.1-21.7v2.1c0 82.9 58.6 151.6 136.7 167.4a180.6 180.6 0 0 1-44.9 5.8c-11.1 0-21.6-1.1-32.2-2.6C211 652 273.9 701.1 348.8 702.7c-58.6 45.9-132 72.9-211.7 72.9-14.3 0-27.5-.5-41.2-2.1C171.5 822 261.2 850 357.8 850 671.4 850 843 590.2 843 364.7c0-7.4 0-14.8-.5-22.2 33.2-24.3 62.3-54.4 85.5-88.2z"
              />
            </svg>
          </button>
          <button
            aria-label="linkedin"
            class="styles__IconButton-sc-12qr9fv-3 c5"
            color="hsla(0, 0%, 100%)"
          >
            <svg
              class="styles__LinkedinIcon-sc-12qr9fv-7"
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM349.3 793.7H230.6V411.9h118.7v381.8zm-59.3-434a68.8 68.8 0 1 1 68.8-68.8c-.1 38-30.9 68.8-68.8 68.8zm503.7 434H675.1V608c0-44.3-.8-101.2-61.7-101.2-61.7 0-71.2 48.2-71.2 98v188.9H423.7V411.9h113.8v52.2h1.6c15.8-30 54.5-61.7 112.3-61.7 120.2 0 142.3 79.1 142.3 181.9v209.4z"
              />
            </svg>
          </button>
          <button
            aria-label="whatsapp"
            class="styles__IconButton-sc-12qr9fv-3 c5"
            color="hsla(0, 0%, 100%)"
          >
            <svg
              class="styles__WhatsAppIcon-sc-12qr9fv-6"
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              t="1569683925316"
              version="1.1"
              viewBox="0 0 1024 1024"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs />
              <path
                d="M713.5 599.9c-10.9-5.6-65.2-32.2-75.3-35.8-10.1-3.8-17.5-5.6-24.8 5.6-7.4 11.1-28.4 35.8-35 43.3-6.4 7.4-12.9 8.3-23.8 2.8-64.8-32.4-107.3-57.8-150-131.1-11.3-19.5 11.3-18.1 32.4-60.2 3.6-7.4 1.8-13.7-1-19.3-2.8-5.6-24.8-59.8-34-81.9-8.9-21.5-18.1-18.5-24.8-18.9-6.4-0.4-13.7-0.4-21.1-0.4-7.4 0-19.3 2.8-29.4 13.7-10.1 11.1-38.6 37.8-38.6 92s39.5 106.7 44.9 114.1c5.6 7.4 77.7 118.6 188.4 166.5 70 30.2 97.4 32.8 132.4 27.6 21.3-3.2 65.2-26.6 74.3-52.5 9.1-25.8 9.1-47.9 6.4-52.5-2.7-4.9-10.1-7.7-21-13z"
              />
              <path
                d="M925.2 338.4c-22.6-53.7-55-101.9-96.3-143.3-41.3-41.3-89.5-73.8-143.3-96.3C630.6 75.7 572.2 64 512 64h-2c-60.6 0.3-119.3 12.3-174.5 35.9-53.3 22.8-101.1 55.2-142 96.5-40.9 41.3-73 89.3-95.2 142.8-23 55.4-34.6 114.3-34.3 174.9 0.3 69.4 16.9 138.3 48 199.9v152c0 25.4 20.6 46 46 46h152.1c61.6 31.1 130.5 47.7 199.9 48h2.1c59.9 0 118-11.6 172.7-34.3 53.5-22.3 101.6-54.3 142.8-95.2 41.3-40.9 73.8-88.7 96.5-142 23.6-55.2 35.6-113.9 35.9-174.5 0.3-60.9-11.5-120-34.8-175.6z m-151.1 438C704 845.8 611 884 512 884h-1.7c-60.3-0.3-120.2-15.3-173.1-43.5l-8.4-4.5H188V695.2l-4.5-8.4C155.3 633.9 140.3 574 140 513.7c-0.4-99.7 37.7-193.3 107.6-263.8 69.8-70.5 163.1-109.5 262.8-109.9h1.7c50 0 98.5 9.7 144.2 28.9 44.6 18.7 84.6 45.6 119 80 34.3 34.3 61.3 74.4 80 119 19.4 46.2 29.1 95.2 28.9 145.8-0.6 99.6-39.7 192.9-110.1 262.7z"
              />
            </svg>
          </button>
          <button
            aria-label="clipboard"
            class="styles__IconButton-sc-12qr9fv-3 c5"
            color="hsla(0, 0%, 100%)"
          >
            <svg
              class="styles__CopyIcon-sc-12qr9fv-8"
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM382 896h-.2L232 746.2v-.2h150v150z"
              />
            </svg>
          </button>
        </div>
        <input
          class="c6 c7"
          readonly=""
          value="http://localhost/"
        />
      </div>
    </div>
  </div>
  <button
    aria-label="share"
    class="c8"
  >
    <svg
      class="styles__Icon-sc-1y03r16-1"
      color="hsla(0, 0%, 52%)"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(133, 133, 133);"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M752 664c-28.5 0-54.8 10-75.4 26.7L469.4 540.8a160.68 160.68 0 0 0 0-57.6l207.2-149.9C697.2 350 723.5 360 752 360c66.2 0 120-53.8 120-120s-53.8-120-120-120-120 53.8-120 120c0 11.6 1.6 22.7 4.7 33.3L439.9 415.8C410.7 377.1 364.3 352 312 352c-88.4 0-160 71.6-160 160s71.6 160 160 160c52.3 0 98.7-25.1 127.9-63.8l196.8 142.5c-3.1 10.6-4.7 21.8-4.7 33.3 0 66.2 53.8 120 120 120s120-53.8 120-120-53.8-120-120-120zm0-476c28.7 0 52 23.3 52 52s-23.3 52-52 52-52-23.3-52-52 23.3-52 52-52zM312 600c-48.5 0-88-39.5-88-88s39.5-88 88-88 88 39.5 88 88-39.5 88-88 88zm440 236c-28.7 0-52-23.3-52-52s23.3-52 52-52 52 23.3 52 52-23.3 52-52 52z"
      />
    </svg>
  </button>
</div>
`;
